osu.Game/Skinning/Editor/SkinSelectionHandler.cs

```diff
@@ -4,22 +4,152 @@
 using System;
 using System.Collections.Generic;
 using System.Linq;
+using System.Runtime.CompilerServices;
 using osu.Framework.Allocation;
+using osu.Framework.Bindables;
 using osu.Framework.Extensions.EnumExtensions;
 using osu.Framework.Graphics;
 using osu.Framework.Graphics.Primitives;
 using osu.Framework.Graphics.UserInterface;
+using osu.Framework.Localisation;
 using osu.Framework.Utils;
 using osu.Game.Extensions;
 using osu.Game.Graphics.UserInterface;
 using osu.Game.Rulesets.Edit;
 using osu.Game.Screens.Edit.Compose.Components;
 using osuTK;
+using Humanizer;
+using osu.Game.Localisation;
 
 namespace osu.Game.Skinning.Editor
 {
     public class SkinSelectionHandler : SelectionHandler<ISkinnableDrawable>
     {
+        /// <summary>
+        /// <p>Keeps track of whether a <see cref="Drawable"/> is using the closest <see cref="Drawable.Anchor">anchor point</see> within its <see cref="Drawable.Parent">parent</see>,
+        /// or whether the user is overriding its anchor point.</p>
+        /// <p>Each <see cref="KeyValuePair{TKey,TValue}.Key">key</see> is either a direct cast of an Anchor value, or it is equal to <see cref="hash_of_closest_anchor_menu_item"/>. This is done
+        /// because the "Closest" menu item is not a valid anchor, so something other than an anchor must be used.</p>
+        /// <p>Each <see cref="KeyValuePair{TKey,TValue}.Value">value</see> is a <see cref="BindableBool"/>. If the <see cref="Bindable{T}.Value"/> is <see langword="false"/>, the user has
+        /// overridden the anchor point.
+        /// If <see langword="true"/>, the closest anchor point is assigned to the Drawable when it is either dragged by the user via <see cref="HandleMovement"/>, or when "Closest" is assigned from
+        /// the anchor context menu via <see cref="applyAnchor"/>.</p>
+        /// </summary>
+        /// <remarks>
+        /// <p>A <see cref="ConditionalWeakTable{TKey,TValue}">ConditionalWeakTable</see> is preferable to a <see cref="Dictionary{TKey,TValue}">Dictionary</see> because a Dictionary will keep
+        /// orphaned references to a Drawable forever, unless manually pruned</p>
+        /// <p><see cref="BindableBool"/> is used as a thin wrapper around <see cref="Boolean">bool</see> because ConditionalWeakTable requires a reference type as both a key and a value.</p>
+        /// </remarks>
+        private readonly ConditionalWeakTable<Drawable, BindableBool> isDrawableUsingClosestAnchorLookup = new ConditionalWeakTable<Drawable, BindableBool>();
```

Comment:
This will need to be stored within `SkinnableInfo` so it can be serialised (as mentioned elsewhere). Just adding a `bool` to track it is fine.

Reply:
Does this mean you prefer `Anchor` + `bool` over `SkinAnchor` throughout?

Reply:
Yeah, I think that will result in better code.

Reply:
Right, the calculated anchor will still have to be stored so that drawables receive their actual anchors even when closest is set, I agree with the `Anchor` + `bool` then.